### DISCUSSION TITLE
kafka suuport volume azureDisk ?

### DISCUSSION THREAD

**Discussion by chanvoyage** [strimzi/strimzi-kafka-operator#5395] [Q&A]
hello everyone 

i have some question about kafka storage

1) if i have existing azure disk , Is it possible we can use  volume azure disk instead of persistent-claim , it look like really cumbersome 

![image](https://user-images.githubusercontent.com/39899497/128217736-03245daf-e662-40fa-8b33-073384d29b06.png)

![image](https://user-images.githubusercontent.com/39899497/128218140-a2cb552a-840e-4791-a052-551564b2ed01.png)

2) which one is better  of solution between  

 "storageclass to provisioning  azure disk for kafka"  and    "attached azure disk to workernode and create local persistent volume"

**Comment by scholzj** [comment]
Not sure I completely follow the question. The usual way to provision storage on Kubernetes is through the persistent volume claims and storage class. If you want to use some specific volume, you can also create the PersistentVolumeClaim manually before deploying Kafka with the right name and it should be re-used by the operator. Keep in mind that using existing volume with some data would require them to be in the right structure etc.
